source/js/main.js

```diff
@@ -293,15 +293,90 @@ function initializeCreateRecipeButtons() {
+ */
+ async function searchButtonHandler() {
+  const query = document.querySelector('.form-control').value;
+  const numOfRecipe = 4;    //number of recipe, 2 for testing
+  const pageOffset = 0; //begin with page 0 when search
+
+  const searchResultPageTitle = document.getElementById('search-results-title');
+  searchResultPageTitle.innerHTML = "Top recipes for " + query;
+  const searchResult = await apiFuncs.getRecipesByName(query, numOfRecipe, pageOffset);
+
+  storageFuncs.storeRecipeData(query, searchResult);
+  //API is down, will add Autocomplete once API work
+  // if(searchResult.length < numOfRecipe){
+  //   const numOfAdditionRecipe = numOfRecipe - searchResult.length;
+  //   const autocompleteResult = await apiFuncs.getRecipesByAutocomplete(query, numOfAdditionRecipe);
+  //   storageFuncs.storeRecipeData(query, autocompleteResult);
+  // }
+   
+  const resultRecipeId = JSON.parse(localStorage.getItem('categories'))[query];
+
+  const searchResultCardslocation = document.querySelector('#results-grid-recipe-card-location');
+  removeAllChildNodes(searchResultCardslocation); //refresh cards
+  createRecipeCards(resultRecipeId, searchResultCardslocation, numOfRecipe);
+}
+
+/**
+ * active when show more button click
+ * add # more recipe cards by fetch # more recipe from API with offset
+ * populate more recipe cards
+ */
+async function searchResultShowMore(){
+  const query = document.querySelector('.form-control').value;
+  const numOfCardExist = document.querySelector('#results-grid-recipe-card-location').childElementCount;
+
+  const numOfAdditionRecipeCards = 4; //2 for testing
+  const pageOffset = numOfCardExist/numOfAdditionRecipeCards; //move to next page 
+  
+  const searchResult = await apiFuncs.getRecipesByName(query, numOfAdditionRecipeCards, pageOffset);
+  storageFuncs.storeRecipeData(query, searchResult);
+
+  const searchResultCardslocation = document.querySelector('#results-grid-recipe-card-location');
+  const numOfRecipe = numOfAdditionRecipeCards + numOfCardExist;    //number of total recipe cards
+
+  const NumOfAllRecipeIdInLocal = JSON.parse(localStorage.getItem('categories'))[query].length;
+
+  //create more cards
+  for(let i = numOfCardExist; (i < numOfRecipe) && (i < NumOfAllRecipeIdInLocal); i++){
+    const singleResultRecipeId = JSON.parse(localStorage.getItem('categories'))[query][i];
+    createRecipeCards([singleResultRecipeId], searchResultCardslocation, 1);
+  }
+}
+
 async function init() {
   initializeRoutes();
   bindPopState();
   populateExplore();
   populateSavedRecipes();
   initializeCreateRecipeButtons();
-
+  
   // temporary code
   testCreateRecipeCards();
+  // localStorage.clear(); //for testing, turn off to make less api call
 }
 
 window.addEventListener('DOMContentLoaded', init);
+
+//search button event listener
+let sButton = document.getElementById('search-results-nav');
+sButton.addEventListener('click', () => searchButtonHandler());
```

Comment:
we don't need these two lines anymore if you move your code into the other function